### ISSUE TITLE
Hyperparameter settings and cold start value head

### ISSUE THREAD

**Issue by huiyeruzhou** [huggingface/trl#1228]
Hello everyone, I am currently using PPOTrainer for training, and some of the hyperparameter settings are making me very confused. I've noticed different terms like batch/minibatch/microbatch. The [document](https://huggingface.co/docs/trl/trainer) doesn't provide detailed explanation so I read some previous PR discussion. It seems that in PPOConfig:

`batch` is the granularity loaded by the dataloader and doesn't seem to play a significant role internally in PPOTrainer.
`minibatch `is the actual granularity of execution, equivalent to what is [discussed](https://github.com/huggingface/trl/pull/546) as microbatch.
`minibatch `* `gradient_accumulation_steps` is the granularity for gradient updates, equivalent to what is [discussed](https://github.com/huggingface/trl/pull/546) as minibatch.

So, if my goal is to achieve a smoother learning curve while keeping memory usage in check, would it be sufficient to increase gradient_accumulation_steps and maintain the sizes of minibatch and batch the same?

Additionally, I've noticed that ValueHead is a cold start layer. I've read the original PPO paper, and there is no mention of pretraining this head. Currently, I am using PPO for code generation training, and due to the task's high sensitivity, I observe that training is struggling to converge and eventually explodes. Would pretraining the Value head help alleviate this issue?

Thanks in advance for any helpful comments~~~

**Comment by lvwerra** [comment]
Yes, your understanding of the batches is correct! Note that at some point you'll have only very few updates per rollout if you increase grad_acc so you might want to also update the batch size of the dataloader.

Indeed, the ValueHead is randomly initialized but I am not sure how you would pretrain it since it's related to the RL loop and the reward function?

**Comment by huiyeruzhou** [comment]
> Yes, your understanding of the batches is correct! Note that at some point you'll have only very few updates per rollout if you increase grad_acc so you might want to also update the batch size of the dataloader.

Cool, thank you for this confirmation!

> Indeed, the ValueHead is randomly initialized but I am not sure how you would pretrain it since it's related to the RL loop and the reward function?

Maybe I should use the word 'warmup' instead of 'pertrained', just freeze the parameter of the pretrained_model by set requires_grad=False and run PPO training for a few epochs, so that the value head can be trained without influence the pretrained_model.

**Comment by lvwerra** [comment]
If you can run some experiments to show that it makes a difference we are happy to consider adding it to the library as an option!
